zonos/model.py

```diff
@@ -302,3 +322,28 @@ def generate(
         self._cg_graph = None  # reset cuda graph to avoid cache changes
 
         return out_codes
+
+import dataclasses
+
+
+class ZonosDecodeOne(torch.nn.Module):
+    def __init__(self, model, inference_params):
+        super().__init__()
+        self.model = model
+        # NB: induce a copy, so mutations don't reflect
+        self.inference_params = dataclasses.replace(inference_params, key_value_memory_dict=None, lengths_per_sample=None, seqlen_offset=None)
+
+    def forward(self, input_ids, key_value_memory_dict, lengths_per_sample, seqlen_offset_box):
```

Comment:
If you want to call this from C++, this is the signature of the module that you have to invoke.